aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -335,12 +335,19 @@ void cdist_backward_kernel_impl(Tensor& result, const Tensor& grad, const Tensor
   const int block_x = 64;
   const int block_y = 16;
   const int grid_x = (m + block_x * 8 - 1) / (block_x * 8);
-  const int grid_y = (dist.numel() + block_y - 1) / block_y;
 
-  const dim3 grid(grid_x, grid_y);
+  const int64_t count = dist.numel();
+  const int64_t grid_temp = (count + block_y - 1) / block_y;
+
+  const int grid_y = int(std::sqrt(float(grid_temp - 1))) + 1;
+  const int grid_z = (grid_temp + grid_y - 1) / grid_y;
+
+  // const int grid_z = grid_temp / 65536 ? 65535 : grid_temp;
```

Comment:
You have to divide by 65535 here. This approach looks cleaner than the previous one, with sqrt.  Something like
```
const auto grid_z = (grid_temp - 1)/65535 + 1;
const auto grid_y = (grid_temp - 1) /grid_z + 1; 
```